Foundation/NSMassFormatter.swift

```diff
@@ -21,28 +20,238 @@ extension MassFormatter {
+        if numberFormatter.locale.sr3202_fix_isMetricSystemLocale() {
+            if numberInKilograms > 1.0 || numberInKilograms <= 0.0 {
+                return .kilogram
+            } else {
+                return .gram
+            }
+        } else {
+            let metricMeasurement = Measurement<UnitMass>(value:numberInKilograms, unit: .kilograms)
+            let imperialMeasurement = metricMeasurement.converted(to: .pounds)
+            let numberInPounds = imperialMeasurement.value
+            
+            if numberInPounds >= 1.0 || numberInPounds <= 0.0  {
+                return .pound
+            } else {
+                return .ounce
+            }
+        }
+    }
+    
+    /// Formats the given value and unit into a string containing one logical 
+    /// value. This is intended for units like kilogram and pound where 
+    /// fractional values are represented as a decimal instead of converted 
+    /// values in another unit.
+    ///
+    /// - Parameter value: The mass's value in the given unit.
+    /// - Parameter unit: The unit used in the resulting mass string.
+    /// - Returns: A properly formatted mass string for the given value and unit.
+    private func singlePartString(fromValue value: Double, unit: Unit) -> String {
+        guard let formattedValue = numberFormatter.string(from:NSNumber(value: value)) else {
+            fatalError("Cannot format \(value) as string")
+        }
+        
+        let separator = unitStyle == MassFormatter.UnitStyle.short ? "" : " "
+        
+        return "\(formattedValue)\(separator)\(unitStringDisplayedAdjacent(toValue: value, unit: unit))"
+    }
+    
+    /// Return the locale-appropriate unit to be shown adjacent to the given
+    /// value.  In most cases this will match `unitStringDisplayedAdjacent(toValue:, unit:)` 
+    /// however there are a few special cases:
+    ///     - Imperial pounds with a short representation use "lb" in the
+    ///       abstract and "#" only when shown with a numeral.
+    ///     - Stones are are singular in the abstract and only plural when 
+    ///       shown with a numeral.
+    ///
+    /// - Parameter value: The mass's value in the given unit.
+    /// - Parameter unit: The unit used in the resulting mass string.
+    /// - Returns: The locale-appropriate unit
+    open func unitStringDisplayedAdjacent(toValue value: Double, unit: Unit) -> String {
+        if unit == .pound && unitStyle == .short {
+            return "#"
+        } else if unit == .stone && unitStyle == .long {
+            if value == 1.0 {
+                return MassFormatter.largeSingularSymbol[unit]!
+            } else {
+                return MassFormatter.largePluralSymbol[unit]!
+            }
+        } else {
+            return unitString(fromValue: value, unit: unit)
+        }
+    }
+    
+
+    
+    /// The number of pounds in 1 stone
+    private static let poundsPerStone = 14.0
```

Comment:
My goal was to use `Measurement<UnitMass>(value:1, unit:.stones).converted(to:.pounds)` here, but that ends up being 0.3471688213196 lb instead of the expected 14 lb.